### PR TITLE
feat: capture HTTP context for API Gateway-triggered Lambda functions

### PR DESCRIPTION
**Pull request by trentm**
https://github.com/elastic/apm/blob/main/specs/agents/tracing-instrumentation-aws-lambda.md#api-gateway--lambda-urls
> The agent should use the information in the request and response
> objects to fill the HTTP context (context.request and context.response)
> fields in the same way it is done for HTTP transactions.

Closes: #2419

# Checklist

- [x] `context.response` as well (initial commit just has `context.request` handling
- [x] changelog entry

### CODE REVIEW

File: test/lambda/trans-setDefaultName.test.js [hunk 1]
```diff
@@ -87,6 +87,7 @@ tape.test('lambda instrumentation does not break trans.setDefaultName', function
         cb(null, 'hi')
       })
 
+      server.clear()
```
Comment (trentm) [inline test/lambda/trans-setDefaultName.test.js:90]: REVIEW NOTE: This was an unrelated fix for this test. It wasn't testing what it thought it was testing and was passing by accident.

**Review by delvedor — APPROVED** [review]
Supporting so many versions is no fun, LGTM!
If it helps, we did some work with lambda in the fastify community, you can see it here: https://github.com/fastify/aws-lambda-fastify
